plugins/dynamix/include/Report.php

```diff
@@ -12,6 +12,16 @@
 ?>
 <?
 switch ($_POST['cmd']) {
+case 'vfat':
+  $check = "/var/tmp/check.vfat";
+  if (!file_exists($check)) {
+    $disks = parse_ini_file('/var/local/emhttp/disks.ini',true);
+    file_put_contents($check,exec("lsblk -lo NAME /dev/{$disks['flash']['device']}|awk '(NR>2)'"));
+  }
+  $dev = file_get_contents($check);
+  passthru("fsck.vfat -n /dev/$dev",$status);
```

Comment:
From my 100% working system with NO issues and is not corrupted at all which would fail your tests

root@unRaidA:/# fsck.vfat /dev/sda
fsck.fat 4.2 (2021-01-31)
Logical sector size (1766 bytes) is not a multiple of the physical sector size.
root@unRaidA:/# echo $?
1

Reply:
Or including the partition info,

root@unRaidA:/# fsck.vfat -n /dev/sda1
fsck.fat 4.2 (2021-01-31)
Dirty bit is set. Fs was not properly unmounted and some data may be corrupt.
 Automatically removing dirty bit.

Leaving filesystem unchanged.
/dev/sda1: 4322 files, 61969/1904448 clusters
root@unRaidA:/# echo $?
1

Now, considering that in *theory* Unraid does an fsck.vfat on bootup (which it's supposed to), this isn't exactly correct, and you can't be flagging a ton of flash drives as being "corrupt" when they're not.